net/httpclient.go

```diff
@@ -21,14 +21,66 @@ const (
 	defaultRefreshInterval = 5 * time.Minute
 )
 
+// Lookuper is an indirection and used to map calls to
+// secrets.SecretsReader's GetSecret to the right parameter.
+type Lookuper interface {
+	// Lookup maps an URL string to the parameter passed into
+	// SecretsReader.GetSecret().
+	Lookup(*url.URL) string
+}
+
+// SingleStaticSecretLookuper stores the string to statically lookup
+type SingleStaticSecretLookuper string
+
+// NewSingleStaticSecretLookuper creates a SingleStaticSecretLookuper
+// that lookups always to the given s.
+func NewSingleStaticSecretLookuper(s string) SingleStaticSecretLookuper {
```

Comment:
NewStaticSecret?

Reply:
TBH, not sure if this initializer is needed at all, it's equivalent to StaticSecret(s)

Reply:
true